File: Neetcode - ALL/Arrays and Hashing/contains-duplicate.cpp

// https://leetcode.com/problems/contains-duplicate/

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, int> mp;

        for(auto ele : nums){
            if(mp.count(ele) > 0) return true;
            mp[ele]++;
        }

        return false;
    }
};

File: Neetcode-150-js/Arrays and Hashing/contains-duplicate.js

// https://leetcode.com/problems/contains-duplicate/

/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const st = new Set();
    for(let i = 0; i < nums.length; i++){
        if(st.has(nums[i])) return true;
        st.add(nums[i]);
    }
    return false;
};